bin/fixmystreet.com/buckinghamshire-flytipping

```diff
@@ -0,0 +1,51 @@
+#!/usr/bin/env perl
+#
+# If a district flytipping report within Buckinghamshire has not been closed
+# after three weeks, close it with a message.
+
+use v5.14;
+use warnings;
+
+BEGIN {
+    use File::Basename qw(dirname);
+    use File::Spec;
+    my $d = dirname(File::Spec->rel2abs($0));
+    require "$d/../../setenv.pl";
+}
+
+use constant BUCKS_AREA_ID => 2217;
+use constant DISTRICT_IDS => (2255, 2256, 2257, 2258);
+use constant TIME_OPEN => '3 weeks';
+
+use FixMyStreet::DB;
+use FixMyStreet::Script::ArchiveOldEnquiries;
+
+my $body = FixMyStreet::DB->resultset("Body")->for_areas(BUCKS_AREA_ID)->first;
+die "Could not find Bucks body" unless $body;
+
+my @districts = FixMyStreet::DB->resultset("Body")->for_areas(DISTRICT_IDS)->all;
+my @district_ids = map { $_->id } @districts;
+die "Did not find all districts" unless @district_ids == 4;
+
+my $template = FixMyStreet::DB->resultset("ResponseTemplate")->search({
+    body => $body, title => 'Auto-closure',
+});
+die "Could not find Bucks Flytipping template" unless $template;
+
+# Fetch all Flytipping problems sent only to districts, older than TIME_OPEN
+my $q = FixMyStreet::DB->resultset("Problem")->to_body(\@district_ids)->search(
+    \[ "NOT regexp_split_to_array(bodies_str, ',') @> ?", [ {} => [ $body->id ] ] ]
```

Comment:
I did not know about @> as an operator.

Reply:
Me neither! `to_body` uses `&&` so returns any that overlap, guess a one-query alternative would be using: `\[ "regexp_split_to_array(bodies_str, ',') = ?", [ {} => [ $body->id ] ] ]` which might be clearer too.

Reply:
would that not get reports only sent to the body?